script.js

```diff
@@ -0,0 +1,245 @@
+    allTasks = copyTasks;
+    renderApp();
+}
+
+function showEditModal(taskId) {
+    const task = allTasks.find(task => task.id === taskId);
+    modalLabelText.textContent = KeysVariables.editTask;
+    submitButton.textContent = KeysVariables.save;
+    inputTitle.value = task.title;
+    inputText.value = task.text;
+    form.elements[task.priority].checked = true;
+    editTaskId = taskId;
+    modalEditMode = true;
+}
+
+
+function tasksBlockListener({target}) {
+    const dropDownMenu = target.closest('.dropdown-menu');
+    if (dropDownMenu) {
+        const taskId = dropDownMenu.getAttribute('id');
+        if (target.classList.contains(KeysVariables.editBtn)) {
+            showEditModal(taskId);
+        } else if (target.classList.contains(KeysVariables.changeStatusBtn)) {
+            changeTaskStatus(taskId);
+        } else {
+            removeTask(taskId);
+        }
+    }
+}
+
+currentTasksBlock.addEventListener('click', tasksBlockListener);
+completedTasksBlock.addEventListener('click', tasksBlockListener);
+
+function closeModal() {
+    modalLabelText.textContent = KeysVariables.addTask;
+    submitButton.textContent = KeysVariables.addTask;
+    form.querySelectorAll('.form-check-input')[1].checked = true;
+    inputTitle.value = '';
+    inputText.value = '';
+    modalEditMode = false;
+}
+
+function getSelectedPriority() {
+    form.querySelectorAll('.form-check-input').forEach((checkbox) => {
+        if (checkbox.checked) {
+            return checkbox.value;
+        }
+    })
+    return null;
+}
+
+function editTask() {
+    const task = allTasks.find(task => task.id === editTaskId);
+    task.priority = getSelectedPriority();
+    task.title = inputTitle.value;
+    task.text = inputText.value;
+}
+
+function getTimeValues(time) {
+    const date = new Date(time);
+    const timeString = date.toTimeString().split('').splice(0, 5).join('');
+    const dateString = date.toLocaleDateString();
+    return {timeString, dateString};
+}
+
+function addTask() {
+    const taskData = {};
+    taskData.priority = getSelectedPriority();
+    taskData.title = inputTitle.value;
+    taskData.text = inputText.value;
+    taskData.id = generateId();
+    taskData.isDone = false;
+    taskData.time = new Date().getTime();
+    allTasks.push(taskData);
+}
+
+function formSubmitListener(event) {
+    event.preventDefault();
+    if (modalEditMode) {
+        editTask();
+    } else {
+        addTask();
+    }
+    closeFormButton.click();
+    updateLocalStorage(LocalStorageKeys.tasks, allTasks);
+    renderApp();
+    closeModal();
+}
+
+form.addEventListener('submit', formSubmitListener);
+
+modal.addEventListener('click', ({target}) => {
+    if (!target.closest('.modal-content')) {
+        closeModal();
+    }
+});
+closeFormButton.addEventListener('click', closeModal);
+closeFormIcon.addEventListener('click', closeModal);
+
+
+function generateId() {
+    return Math.random().toString(36).slice(2);
+}
```

Comment:
organise you code a little bit. if you don't use modules than locate all your helpers like generateId, getTimeValues and etc in one place. file is a little messy